pro/src/components/pages/Offerers/Offerer/VenueV1/VenueCreation/VenueCreation.jsx

```diff
@@ -27,22 +37,59 @@ import LocationFields, {
   FRANCE_POSITION,
 } from '../fields/LocationFields'
 import WithdrawalDetailsFields from '../fields/WithdrawalDetailsFields'
+import NotificationMessage from '../Notification'
+import { formatVenuePayload } from '../utils/formatVenuePayload'
+import VenueLabel from '../ValueObjects/VenueLabel'
+import VenueType from '../ValueObjects/VenueType'
 
+/* eslint-disable */
+const withRouter = Component => {
+  return props => (
+    <Component
+      {...props}
+      currentUser={useSelector(state => selectCurrentUser(state))}
+      dispatch={useDispatch()}
+      history={useHistory()}
+      isBankInformationWithSiretActive={useActiveFeature(
+        'ENFORCE_BANK_INFORMATION_WITH_SIRET'
+      )}
+      isEntrepriseApiDisabled={useActiveFeature('DISABLE_ENTERPRISE_API')}
+      location={useLocation()}
+      params={useParams()}
+    />
+  )
+}
+/* eslint-enable */
+
+// TO FIX:: the above component is temporary, in order to
+// allow react-router upgrade before the below component is refactored as function component
+// eslint-disable-next-line react/no-multi-comp
 class VenueCreation extends PureComponent {
-  constructor() {
+  constructor(props) {
     super()
+    this.handleInitialRequest = this.handleInitialRequest.bind(this)
```

Comment:
J'ai du mal à comprendre l'utilité de cette manipulation.

Reply:
c'est pour que le this soit bien le composant quand la fonction de callback est appelée

Reply:
C'est à ca que serre `bind()` mais ici tu prend une méthode de l'objet et tu lui redonne l'objet en context.
Testé en local sans cette boucle de bind et ca fonctionne bien.

Reply:
ok j'avais peur de provoquer des side-effects, si tu es sur de toi je les enlève